zkevm-circuits/src/evm_circuit/execution/error_oog_static_memory.rs

```diff
@@ -92,45 +167,155 @@ impl<F: Field> ExecutionGadget<F> for ErrorOOGStaticMemoryGadget<F> {
         offset: usize,
         block: &Block<F>,
         _: &Transaction,
-        _: &Call,
+        call: &Call,
         step: &ExecStep,
     ) -> Result<(), Error> {
         let opcode = step.opcode.unwrap();
 
+        println!("{}", step.rw_indices.len());
+
         // Inputs/Outputs
         let address = block.rws[step.rw_indices[0]].stack_value();
         self.address
             .assign(region, offset, Some(address.to_le_bytes()))?;
 
-        // Check if this is an MSTORE8
-        let is_mstore8 = self.is_mstore8.assign(
+        self.opcode
+            .assign(region, offset, Value::known(F::from(opcode.as_u64())))?;
+        self.is_mload.assign(
+            region,
+            offset,
+            F::from(opcode.as_u64()),
+            F::from(OpcodeId::MLOAD.as_u64()),
+        )?;
+        self.is_mstore8.assign(
             region,
             offset,
             F::from(opcode.as_u64()),
             F::from(OpcodeId::MSTORE8.as_u64()),
         )?;
 
         // Address in range check
-        self.address_in_range.assign(
+        let address_high = address_high::value::<F>(address.to_le_bytes());
+        self.address_in_range_high
+            .assign(region, offset, address_high)?;
+
+        let address_value = address_low::value(address.to_le_bytes());
+        let size = if opcode == OpcodeId::MSTORE8 { 1 } else { 32 };
+
+        let expanded_address = address_value
+            .checked_add(size)
+            .expect("address overflow u64");
+        self.expanded_address_in_range.assign(
             region,
             offset,
-            address_high::value::<F>(address.to_le_bytes()),
+            F::from(expanded_address),
+            F::from(1u64 << (N_BYTES_MEMORY_ADDRESS * 8)),
         )?;
 
-        // Memory expansion
-        self.memory_expansion.assign(
+        // TODO: sanity check, remove this after fixing #347 missing ErrGasUintOverflow
+        if address_high != F::zero() || expanded_address > (1u64 << (N_BYTES_MEMORY_ADDRESS * 8)) {
+            panic!("address overflow {} bytes", N_BYTES_MEMORY_ADDRESS);
+        }
+
+        let (_, memory_cost) = self.memory_expansion.assign(
             region,
             offset,
             step.memory_word_size(),
-            [address_low::value(address.to_le_bytes())
-                + if is_mstore8 == F::one() { 1 } else { 32 }],
+            [expanded_address],
         )?;
 
         // Gas insufficient check
         // Get `gas_available` variable here once it's available
-        self.insufficient_gas
-            .assign(region, offset, F::from(step.gas_cost - step.gas_left))?;
+        self.insufficient_gas.assign(
```

Comment:
here you can use gas cost from step as following :
 ```
  self.insufficient_gas.assign(
            region,
            offset,
            F::from(step.gas_left),
            F::from(step.gas_cost),
        )?;
```

Reply:
step.gas_cost does not contain the expansion cost